src/App.tsx

```diff
@@ -1,26 +1,61 @@
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from './hooks/useAppSelector';
+import { todosSlice } from './features/todos';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  const dispatch = useDispatch();
+  const todos = useAppSelector(state => state.todos);
+  const selectedTodoId = useAppSelector(state => state.currentTodo);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(todosFromServer => {
+        dispatch(todosSlice.actions.loadTodos(todosFromServer));
+        setFilteredTodos(todos);
+      })
+      .finally(() => setIsLoading(false));
+  }, [todos]);
```

Comment:
Including `todos` in the dependency array of `useEffect` causes the effect to run every time `todos` changes. This can lead to an infinite loop because `dispatch(todosSlice.actions.loadTodos(todosFromServer))` updates `todos`, triggering the effect again. Consider using an empty dependency array `[]` if you want the effect to run only once on component mount, or adjust the logic to prevent unnecessary re-renders.